docs: add new API documentation

asaadam opened the pull request:
## Description

Add new API documentation in readme

resir014 reviewed (changes requested):
@asaadam If this is going an we split this into another Markdown document and link to it from the Readme?

Let's create a `docs/api-reference.md` file and move this entire API docs there.

asaadam commented:
> @asaadam If this is going to be long, can we split this into another Markdown document and link to it from the Readme?
> 
> Let's create a `docs/api-reference.md` file and move this entire API docs there.



> @asaadam If this is going to be long, can we split this into another Markdown document and link to it from the Readme?
> 
> Let's create a `docs/api-reference.md` file and move this entire API docs there.

it's a good idea.
On it

ramarahmanda commented:
Refer to this issue https://github.com/kawalcovid19/silacak/issues/20

Can you add this curl to api docs?

Check account status
```
curl 'https://silacak.kemkes.go.id/api/users?fields=displayName%2Cid%2Caccess%2CuserCredentials%5Busername%2Cdisabled%2ClastLogin%2CtwoFA%5D%2CteiSearchOrganisationUnits%5Bid%2Cpath%5D&filter=userCredentials.username:eq:<username>&fields=id' \
  -H 'authority: silacak.kemkes.go.id' \
  -H 'pragma: no-cache' \
  -H 'cache-control: no-cache' \
  -H 'sec-ch-ua: "Chromium";v="92", " Not A;Brand";v="99", "Google Chrome";v="92"' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'authorization: Basic YWRtaW4tcmVnaXN0ZXI6S2Vta2VzMjAyMSUl' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36' \
  -H 'origin: https://reg-silacak.kemkes.go.id' \
  -H 'sec-fetch-site: same-site' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-dest: empty' \
  -H 'referer: https://reg-silacak.kemkes.go.id/' \
  -H 'accept-language: en-US,en;q=0.9' \
  --compressed
```

resir014 commented on docs/api-reference.md:
```diff
@@ -0,0 +1,481 @@
+## API
+
+
+# SILACAK API DOCS
```
```suggestion
# API

## Silacak API Docs
```

- Nitpick: Please use title case instead of all caps for Markdown headings.
- Extra line removed.

Please also re-check all headings and make sure they are at the correct level (h1-h4), for accessibility reasons.

resir014 commented on docs/api-reference.md:
```diff
@@ -0,0 +1,481 @@
+
+#### Base URL : `https://silacak.kemkes.go.id`
+
+## Login
+Authorization: Basic base64encode(username:password)
```
Please give an extra line between all headings and paragraphs.

resir014 commented on docs/api-reference.md:
````diff
@@ -0,0 +1,481 @@
+    /api/organisationUnits?paging=true&fields=displayName,parent,id,level&pageSize=10
+
+### Example HTTP Response
+```
+{
+	"pager": {
+	"page": 1,
+	"pageCount": 994,
+	"total": 9933,
+	"pageSize": 2,
+	"nextPage": "https://training-silacak.kemkes.go.id/api/organisationUnits?page=2&pageSize=10&paging=true&fields=displayName%2Cparent%2Cid%2Clevel"
+},
+"organisationUnits": [
+		{
+			"level": 3,
+			"id": "MVPPtzjrsTp",
+			"displayName": "1101 - KAB. SIMEULUE",
+			"parent": {
+			"id": "nBOW2TBc1mn"
+		},
+		{
+			"level": 3,
+			"id": "TYbR3qLC9aX",
+			"displayName": "1102 - KAB. ACEH SINGKIL",
+			"parent": {
+			"id": "nBOW2TBc1mn"
+		}
+	]
+},
+```
````
Please convert to spaces.

resir014 commented on docs/api-reference.md:
```diff
@@ -0,0 +1,481 @@
+                    "children": true,
+                    "displayName": "KEC. JAGAKARSA",
+                    "parent": {
+			                     "id": "CWKkrIzNOhX"
+			                  }
+                },
+	           ]
+		     }
+		   ]
+		 }
```
Please convert to spaces.

resir014 commented on README.md:
```diff
@@ -60,3 +60,6 @@ See our contribution guidelines in these languages:
 
 When contributing to our project, please use English when communicating with other people in issues and/or pull requests. [Click here](CONTRIBUTING.md#why-are-we-using-english-in-our-issues--prs) to read why. ([Bahasa Indonesia](CONTRIBUTING_ID.md#mengapa-kita-menggunakan-bahasa-inggris-dalam-menulis-issue-dan-pull-request))
+
+## API
+for api documentation, [click here](docs/api-reference.md)
```
```suggestion
## API

For API documentation, [click here](docs/api-reference.md).
```

As mentioned earlier, please give an extra line between headings and paragraphs.

resir014 reviewed (changes requested):
Mostly documentation formatting issues. A clean codebase also comes with clean documentation too.

asaadam commented:
> Refer to this issue #20
> 
> Can you add this curl to api docs?
> 
> Check account status
> 
> ```
> curl 'https://silacak.kemkes.go.id/api/users?fields=displayName%2Cid%2Caccess%2CuserCredentials%5Busername%2Cdisabled%2ClastLogin%2CtwoFA%5D%2CteiSearchOrganisationUnits%5Bid%2Cpath%5D&filter=userCredentials.username:eq:<username>&fields=id' \
>   -H 'authority: silacak.kemkes.go.id' \
>   -H 'pragma: no-cache' \
>   -H 'cache-control: no-cache' \
>   -H 'sec-ch-ua: "Chromium";v="92", " Not A;Brand";v="99", "Google Chrome";v="92"' \
>   -H 'accept: application/json, text/plain, */*' \
>   -H 'authorization: Basic YWRtaW4tcmVnaXN0ZXI6S2Vta2VzMjAyMSUl' \
>   -H 'sec-ch-ua-mobile: ?0' \
>   -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36' \
>   -H 'origin: https://reg-silacak.kemkes.go.id' \
>   -H 'sec-fetch-site: same-site' \
>   -H 'sec-fetch-mode: cors' \
>   -H 'sec-fetch-dest: empty' \
>   -H 'referer: https://reg-silacak.kemkes.go.id/' \
>   -H 'accept-language: en-US,en;q=0.9' \
>   --compressed
> ```
just want to make it clear.
so, do we need to add all curl for each endpoint?

asaadam commented:
@ramarahmanda please help to check this PR. Thanks

zainfathoni commented on README.md:
```diff
@@ -60,3 +60,7 @@ See our contribution guidelines in these languages:
+
+## API
+
+for api documentation, [click here](docs/api-reference.md)
```
```suggestion
Our API documentation is in [api-reference.md](docs/api-reference.md) file.
```

Rephrase the sentence to make it more obvious.

asaadam replied:
Done, please check it agian.

ramarahmanda commented:
@asaadam after formatting issues have been resolved, I think it's good enough. I'll add & improve the information in API after merge. wdyt @zainfathoni @resir014

asaadam commented:
hey, sorry I miss understood your comment @ramarahmanda .
Already add curl for check account status.

zainfathoni reviewed (approved):
LGTM 💯 

Thanks for working on it, @asaadam! 🙏

zainfathoni commented:
@all-contributors please add @asaadam for doc.
